Backend/Remora.Discord.API/API/Objects/Invites/InviteWithMetadata.cs

```diff
@@ -0,0 +1,48 @@
+//
+//  InviteWithMetadata.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+using Remora.Discord.API.Abstractions.Objects;
+using Remora.Rest.Core;
+
+namespace Remora.Discord.API.Objects;
+
+/// <inheritdoc cref="IInviteWithMetadata" />
+public record InviteWithMetadata
+(
+    string Code,
+    Optional<IPartialGuild> Guild,
+    IPartialChannel? Channel,
+    int Uses,
+    int MaxUses,
+    int MaxAge,
+    bool IsTemporary,
```

Comment:
This is missing the json property name configuration (should be `temporary` in json, as per the docs)